File: mobile/library/swift/stats/Tags.swift
import Foundation

/// Class used to represent tags data structures.
/// To instantiate new instances, see `TagsBuilder`.
@objcMembers
public class Tags: NSObject {
  let tags: [String: String]

  /// Get the value for the provided tag name.
  ///
  /// - parameter name: tag name for which to get the current value.
  ///
  /// - returns: The current tag value specified for the provided name.
  public func value(forName name: String) -> String? {
    return self.tags[name]
  }

  /// Accessor for all underlying tags as a map.
  ///
  /// - returns: The underlying tags.
  public func allTags() -> [String: String] {
    return self.tags
  }

  /// Internal initializer used by builders.
  ///
  /// - parameter tags: Tags to set.
  required init(tags: [String: String]) {
    self.tags = tags
    super.init()
  }

  /// Internal initializer used by builders.
  override required init() {
    self.tags = [String: String]()
    super.init()
  }
}

// MARK: - Equatable

extension Tags {
  public override func isEqual(_ object: Any?) -> Bool {
    return (object as? Self)?.tags == self.tags
  }
}

// MARK: - CustomStringConvertible

extension Tags {
  public override var description: String {
    return "\(type(of: self)) \(self.tags.description)"
  }
}

File: mobile/library/kotlin/io/envoyproxy/envoymobile/stats/Tags.kt
package io.envoyproxy.envoymobile

/*
 * Base class that is used to represent tags structures.
 * To instantiate new instances, see `TagsBuilder`.
 */
class Tags {
  @Suppress("MemberNameEqualsClassName") val tags: Map<String, String>

  /**
   * Internal constructor used by builders.
   *
   * @param tags: tags to set.
   */
  internal constructor(tags: Map<String, String>) {
    this.tags = tags
  }

  /**
   * Get the value for the provided tag name.
   *
   * @param name: Tag name for which to get the current value.
   * @return String?, The current tags specified for the provided name.
   */
  fun value(name: String): String? {
    return tags[name]
  }

  /**
   * Accessor for all underlying tags as a map.
   *
   * @return Map<String, String>, The underlying tags.
   */
  fun allTags(): Map<String, String> {
    return tags
  }
}
